primefaces/src/main/java/org/primefaces/model/filter/ComparableFilterConstraint.java

```diff
@@ -38,9 +34,14 @@ public boolean isMatching(FacesContext ctxt, Object value, Object filter, Locale
         if (value == null || filter == null) {
             return false;
         }
+        if (value.getClass() == filter.getClass() && value instanceof Comparable) {
+            return getPredicate().test((Comparable) value, (Comparable) filter);
+        }
         if (value instanceof Number) {
-            BigDecimal filterBigDecimal = toBigDecimal(filter, locale);
```

Comment:
should we do this check in general, not just for Number?
if (value.class not assignable to filter.class)?

Reply:
I do on line 37
```java
        if (value.getClass() == filter.getClass() && value instanceof Comparable) {
            return getPredicate().test((Comparable) value, (Comparable) filter);
        }
```
I could change it to an assignable check

Reply:
Ah.. to show an exception you mean. Yes!